qiime2/metadata/metadata.py

```diff
@@ -899,8 +815,8 @@ def filter_ids(self, ids_to_keep):
         """
         filtered_df = self._filter_ids_helper(self._dataframe, self.get_ids(),
                                               ids_to_keep)
-        filtered_md = self.__class__(filtered_df)
-        filtered_md._add_artifacts(self.artifacts)
+        filtered_md = self.__class__(filtered_df, self._column_names,
```

Comment:
Do we want `filter_ids` and `filter_columns` to remove the columns from the filtered metadata's `column_names` and `column_sources`? Removal from `column_sources` would potentially remove artifacts from provenance. If this metadata is the result of a merge, and all columns from one of the metadatas merged in were filtered out, that metadata's `_source_artifact` would no longer be tracked.

Reply:
I think that retroactively altering the provenance of the metadata is a dangerous game to play, maybe better to track in some other property for when these things are filtered out?